collector/collector.go

```diff
@@ -80,15 +80,17 @@ type nodeCollector struct {
 }
 
 // NewNodeCollector creates a new NodeCollector
-func NewNodeCollector() (*nodeCollector, error) {
+func NewNodeCollector(filters map[string]bool) (*nodeCollector, error) {
```

Comment:
This is not a clean interface, leaking implementation internals (for performance reasons or to safe some lines of code I suppose?) into the function signature.

I propose to use `NewNodeCollector(filters ...string) (*nodeCollector, error)`.